### DISCUSSION TITLE
Shorthand OR statment

### DISCUSSION THREAD

**Discussion by luckyexpert** [dotnet/csharplang#4042] [Language Ideas] [answered]
Feature Idea: Shorthand if - or statement. 

Instead of 
`if(x == 1 || x == 2 || x == 3)...`

you could instead say  
`if(x == (1,2,3))... `
or
`if(x == 1 || 2 || 3)...`
or
`if(x in (1,2,3)...`

I see many people asking this question on Stack Overflow, and even had this discussion with coworkers.. It's not a big deal when variables and values are as short as my example, but consider:
`if(invoice.customer.phone.type == PhoneType.Home || invoice.customer.phone.type == PhoneType.Work|| invoice.customer.phone.type == PhoneType.Cell)...`

I often find myself REALLY wishing there were a quick shorthand method that doesn't involve declaring a new list or array:
`if(invoice.customer.phone.type == PhoneType.Home || PhoneType.Work|| PhoneType.Cell)...`

Forgive me if this has been mentioned before, I'm sure it has but couldn't find any mention of it. This seems like a simple situation most developers beginner to expert would commonly find themselves in.

**Comment by HaloFour** [comment, marked as answer]
With disjunctive patterns in pattern matching in C# 9.0 you can do this:

```cs
if (x is 1 or 2 or 3) { ... }
```

Although you are limited to constants as the operands.

**Reply by luckyexpert** [reply]
wow this is pretty slick, this may work for me most of the time when I am working with constants..

**Comment by spydacarnage** [comment]
Duplicate of [#420](https://github.com/dotnet/csharplang/issues/420)

**Reply by luckyexpert** [reply]
Ah, knew this surely had to have come up before. Thanks for linking

**Comment by spydacarnage** [comment]
And if you need non-constants, there is the option for an extension method
which works wonders.
